### ISSUE TITLE
Error on backup: "manifest validation failed: repo size must be > 0"

### ISSUE THREAD

**Issue by nLassimonneAdveez** [pgbackrest/pgbackrest#2358]
Please provide the following information when submitting an issue (feature requests or general comments can skip this):

1. pgBackRest version:
**2.51**

2. PostgreSQL version:
**16.3**

3. Operating system/version - if you have more than one server (for example, a database server, a repository host server, one or more standbys), please specify each:
**Ubuntu 22.04**

4. Did you install pgBackRest from source or from a package?
**From package**

5. Please attach the following as applicable:
Logs from _main-backup.log_
2024-05-23 04:00:03.928 P00   INFO: last backup label = 20240519-040005F_20240522-072911I, version = 2.51
2024-05-23 04:00:03.928 P00   INFO: execute non-exclusive backup start: backup begins after the requested immediate checkpoint completes
2024-05-23 04:00:13.041 P00   INFO: backup start archive = 000000020001BF7D00000020, lsn = 1BF7D/2001D4D0
2024-05-23 04:00:13.041 P00   INFO: wait for replay on the standby to reach 1BF7D/2001D4D0
2024-05-23 04:00:21.946 P00   INFO: replay on the standby reached 1BF7D/2001D4D0
2024-05-23 04:00:21.946 P00   INFO: check archive for prior segment 000000020001BF7D0000001B
2024-05-23 04:10:33.555 P00   INFO: execute non-exclusive backup stop and wait for all WAL segments to archive
2024-05-23 04:10:33.760 P00   INFO: backup stop archive = 000000020001BF7E0000006C, lsn = 1BF7E/6CB21B60
2024-05-23 04:10:33.939 P00   INFO: check archive for segment(s) 000000020001BF7D00000020:000000020001BF7E0000006C
2024-05-23 04:24:24.733 P00   INFO: new backup label = 20240519-040005F_20240523-040003I
2024-05-23 04:24:24.734 P00  ERROR: [029]: manifest validation failed:
                                    repo size must be > 0 for file 'pg_data/base/18073/16435447_fsm'
2024-05-23 04:24:24.734 P00   INFO: backup command end: aborted with exception [029]

7. Describe the issue:
When doing incremental backup from replica, I got the described error on my second incremental backup, following this pattern:
- Full Backup: OK
- Incremental Backup: OK
- Incremental Backup: Failed
- Full Backup: OK
- Incremental Backup: OK
- Incremental Backup: Failed

My Stanza is stored on a S3 Bucket and is encrypted.

Thanks.

**Comment by dwsteele** [comment]
Can you share your configuration so we can try an reproduce?

**Comment by nLassimonneAdveez** [comment]
Thanks for you responses,
Here is my config file, redacted for privacy:

[main]
pg1-path=/var/lib/postgresql/16/main
pg2-host=[REDACTED]
pg2-host-user=postgres
pg2-path=/var/lib/postgresql/16/main
recovery-option=primary_conninfo=host=[REDACTED] port=5432 user=replication password=[REDACTED]application_name=replica

[global]
repo1-path=[REDACTED]
repo1-retention-full=2
repo1-s3-bucket=[REDACTED]
repo1-s3-endpoint=[REDACTED]
repo1-s3-key=[REDACTED]
repo1-s3-key-secret=[REDACTED]
repo1-s3-region=[REDACTED]
repo1-type=s3
repo1-cipher-pass=[REDACTED]
repo1-cipher-type=aes-256-cbc
repo1-bundle=y
repo1-block=y
process-max=24
backup-standby=y
start-fast=y
archive-check=y
archive-copy=y
archive-async=y
archive-get-queue-max=10GiB

Here is an example of command used for backup (from the log):
2024-05-20 11:24:20.598 P00   INFO: backup command begin 2.51: --archive-check --archive-copy --backup-standby --exec-id=1348535-19dbde07 --log-level-console=debug --pg2-host=<redacted> --pg2-host-user=<redacted> --pg1-path=/var/lib/postgresql/16/main --pg2-path=/var/lib/postgresql/16/main --process-max=24 --repo1-block --repo1-bundle --repo1-cipher-pass=<redacted> --repo1-cipher-type=aes-256-cbc --repo1-path=<redacted> --repo1-retention-full=2 --repo1-s3-bucket=<redacted> --repo1-s3-endpoint=<redacted> --repo1-s3-key=<redacted> --repo1-s3-key-secret=<redacted> --repo1-s3-region=<redacted> --repo1-type=s3 --stanza=main --start-fast

**Comment by dwsteele** [comment]
OK, thanks. I would also be good to get some info on this file. Can you provide the output from:
```
ls -lah /var/lib/postgresql/16/main/base/18073/16435447_fsm
```

**Comment by dwsteele** [comment]
Also, let's get that output from the primary and the replica. Thanks!

**Comment by nLassimonneAdveez** [comment]
Here it is:
**Primary:**
![image](https://github.com/pgbackrest/pgbackrest/assets/87472901/fcce662a-0a0b-4772-a260-5fd550eebf56)
**Replica:**
![image](https://github.com/pgbackrest/pgbackrest/assets/87472901/ddfb569d-09d5-4ada-bf94-51a9bfa447f6)

The issue occured with other files, I identifiied an index and try to remove it and re create it. The issue was gone but it came back with another file.
Thanks.

**Comment by dwsteele** [comment]
Thank you for providing the requested information. I was able to reproduce the bug and have fixed it in #2367.

How would you like your name to appear in the release notes?

**Comment by nLassimonneAdveez** [comment]
Many thanks David !
Yes, if you can mention my name it would be great.

Have a good day.
________________________________
De : David Steele ***@***.***>
Envoyé : vendredi, mai 31, 2024 9:13:49 AM
À : pgbackrest/pgbackrest ***@***.***>
Cc : Nicolas LASSIMONNE ***@***.***>; Author ***@***.***>
Objet : Re: [pgbackrest/pgbackrest] Error on backup: "manifest validation failed: repo size must be > 0" (Issue #2358)


Closed #2358<https://github.com/pgbackrest/pgbackrest/issues/2358> as completed.

**Comment by threenotrump** [comment]
Hi @dwsteele 

Is there a workaround for this bug until the fix comes?

**Comment by dwsteele** [comment]
> Is there a workaround for this bug until the fix comes?

At this time the only workaround that I can think of is to use only diff backups rather than incr. They are a bit larger but you can keep fewer of them, so space-wise it should work out.

**Comment by dwsteele** [comment]
@nLassimonneAdveez This fix has been released in 2.52.1. The Debian package has been prepared (https://salsa.debian.org/postgresql/pgbackrest/-/commit/542e38eeb0b03100aab11b538defda2806185fde) so hopefully that will be released before too long.

**Comment by Thetortuga** [comment]
Thanks very much @dwsteele ! I was wondering if it will be released in 2.53 or before. It is great !

**Comment by dwsteele** [comment]
@Thetortuga The original plan was to release it in 2.53 but that release has a lot of new stuff, so it seemed safer to release the patch separately.
